### DISCUSSION TITLE
CefSharp conflicts in my project with an ActiveX control I also use

### DISCUSSION THREAD

**Discussion by ghost** [cefsharp/CefSharp#3526] [General]
I was wondering if anyone had run into this before. I have a WinForms application that uses both CefSharp and another 3rd party ActiveX control.  We recently had to update the version of the 3rd party control and it appears that it now has some custom built version of CefSharp inside of it.  I see a number of the cefsharp files in the control's installation folder.

I'm trying to use the 89.0.17 version of the CefSharp nuget packages.  My problem is this: I have a form that loads both the CefSharp and the other 3rd party control.  Whichever one of them manages to init the CefSharp stuff first causes the other one to throw exceptions.  This happens because they are clearly different versions of the CefSharp code.  However because of the way it was incorporated into the 3rd party control I can't tell what version they are using.

I'm wondering if there is a way to isolate the two from each other completely but continue to have them both being used within my same form.

**Comment by amaitland** [comment]
> I'm wondering if there is a way to isolate the two from each other completely but continue to have them both being used within my same form.

You can only load one version of CEF within a process. To use multiple versions you'd need to run one out of processes. 

> I see a number of the cefsharp files in the control's installation folder.

You should be able to tell the version from one of the dlls.

I'd be contacting the ActiveX control developer and asking for a version that 
- uses the latest CefSharp version 
- include additional method that allows for explicitly calling Cef.Initialize with additional settings.

**Reply by ghost** [reply]
Well ... poop. Thanks for the info.
